### ISSUE TITLE
Alt Text Inaccessible in Self-Contained HTML Documents

### ISSUE THREAD

**Issue by NV-Codes** [jgm/pandoc#9525]
## Issue

I would like to create a self-contained HTML document that happens to have images from Pandoc's standard Markdown. The `--embed-resources` flag seems to produce a self-contained document lacking alt text; omitting this option yields an HTML document with alt text, as expected. This behavior is consistent whether or not the `-s` flag is used.

The issue presents an accessibility problem for self-contained HTML files generated using Pandoc.

Markdown Example:

```markdown
# File with Image

![A description of my_image](my_image.svg) <!-- Optional comment to control whether an in-line image is produced -->

```

Example Command Used:

```
pandoc -f markdown -t html --embed-resources test_case.md -o test_case.html
```

## Software Version

Pandoc 3.1.12.1 on Windows 11

**Comment by jgm** [comment]
Here's what I see when I try this:
```
<p><svg id="svg2" alt="A description of my_image" width="191.56267" height="151.71201" viewBox="0 0 191.56267 151.71201" xmlns:svg="http://www.w3.org/2000/svg">
 ```
Note the alt attribute...

**Comment by NV-Codes** [comment]
When I inspect the source of the resulting HTML document, I also notice the 'alt' attribute. Perhaps this is an issue with my screen reader, [NVDA](url), as it did not verbalize anything in the self-contained document if my_image.svg was successfully embedded into the document. I am not sure what the difference is to the screen reader.

Thank you for the prompt response!

**Comment by NV-Codes** [comment]
Using Narrator, the built-in screen reader on Windows, I am likewise unable to read alt text from images in self-contained HTML documents.

**Comment by NV-Codes** [comment]
I have done some research, and it seems that embedding the inline SVG code using the `<img>` tag and its corresponding 'alt' attribute should likely solve the problem, as screen readers seem to have near universal support for identifying alt text in `<img>` elements (but apparently not necessarily in others).

I am not experienced with HTML, but I presume the solution for other image formats would be analogous.

I hope this would work to make Pandoc's self-contained HTML output more accessible. [HTML](https://developer.mozilla.org/en-US/docs/Learn/Accessibility/HTML) (and [MathML](https://www.chhs.colostate.edu/accessibility/best-practices-how-tos/mathml/)) are currently some of the best formats for accessibility, and Pandoc greatly facilitates their use.

**Comment by jgm** [comment]
> embedding the inline SVG code using the `<img>` tag and its corresponding 'alt' attribute should likely solve the problem

You mean using a data URI?  This is less elegant than using an svg tag, but may be called for if screen readers can't handle alt on the svg.

**Comment by jgm** [comment]
What about the suggestion here?
https://stackoverflow.com/questions/4697100/accessibility-recommended-alt-text-convention-for-svg-and-mathml

The 2021 advice offered there says:

> For inline `<svg>`'s

> - Add a `<title>` element; this will be included as a visual tooltip, and available to the Accessability API.
> - On the `<svg>` tag, add role="img".
> ```
> <svg role="img">
>  <title>Descriptive Title Here</title>
>  <use xlink:href="#some-icon"></use>
> </svg>
> ```

**Comment by NV-Codes** [comment]
Yes, I just created a sample HTML file with that code (except for `<use xlink:href="#some-icon"></use>`, as I did not have any icon to use), and my screen reader was able to 1) correctly identify the element as a graphic and 2) read 'Descriptive Title Here' upon encountering it.

Would the same general solution work for all supported image formats?

**Comment by jgm** [comment]
I think other image formats, besides svg, just use img tags, so they should be fine.

**Comment by NV-Codes** [comment]
That sounds good, thanks!

**Comment by jgm** [comment]
Can you test if it still works if instead of title we use desc?
```
<svg role="img">
 <desc>Description of image here</desc>
 <use xlink:href="#some-icon"></use>
</svg>
```
That would be better, as it wouldn't cause popup text on hover.

**Comment by NV-Codes** [comment]
That seems only to work when using Firefox to open the file. It seems that Chromium does not have the same support for the `<desc>` tag, at least in terms of forwarding its contents to the relevant accessibility API. I've tested with NVDA and Narrator.

Maybe the 'alt' attribute would work if `role="img"` were also specified. I will test this next.

**Comment by jgm** [comment]
What about removing the title and desc and putting an `aria-label` attribute on the svg?  Does that work?

**Comment by NV-Codes** [comment]
Great, thank you!

**Comment by jgm** [comment]
I forgot to add `role="img"`. But is it needed? Can you test to see if aria-label suffices without it?

**Comment by NV-Codes** [comment]
It does not work without the 'role' attribute.

Inaccessible SVG:

```html
<svg aria-label="Description of image here">
  <use xlink:href="#some-icon"></use>
</svg>
```
